Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1271,18 +1271,37 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
             return;
         }
 
-        var treeOutline = this.treeOutline;
-        var wasExpanded = this.expanded;
+        let treeOutline = this.treeOutline;
+        let wasExpanded = this.expanded;
+        let [newTagName, ...attributes] = newText.trim().split(" ");
 
-        function changeTagNameCallback(error, nodeId)
-        {
+        let changeTagNameCallback = (error, nodeId) => {
             if (error || !nodeId) {
                 cancel();
                 return;
             }
 
-            var node = WI.domManager.nodeForId(nodeId);
+            let node = WI.domManager.nodeForId(nodeId);
+
+            if (attributes.length) {
+                let attributesAddedCallback = (error, nodeId) => {
+                    if (error) {
+                        cancel();
+                        return;
+                    }
+
+                    nodeUpdatedCallback(node);
+                }
+
+                const attributeNameToReplace = "";
+                node.setAttribute(attributeNameToReplace, attributes.map(attr => attr.trim()).join(" "), attributesAddedCallback);
+            } else {
+                const moveFocus = true;
+                nodeUpdatedCallback(node, moveFocus);
+            }
+        }
 
+        let nodeUpdatedCallback = (node, moveFocus) => {
```

Comment:
let's make this into `(node, options = {}) => {` so that we don't need to have a `const` at the callsite